### PR TITLE
Update RELEASING.md to account for GitHub release creation by the CLI pipeline

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Explain manual step after release is created by CI
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -146,6 +146,15 @@ Briefly speaking, it requires executing of the following steps:
...
     >Merging the release PR later allows easier adjusting of the tag to include the fix for the potential issues.
+1. If a GitHub release is automatically created by the CI, as visible on https://github.com/IntersectMBO/cardano-api/releases,
+   undraft the release by clicking the pen on the top right, then untick _Set as a pre-release_, and
+   finally select _Update release_.
+
+   >:warning: **GitHub bug**
+   >
+   > If you try to undraft a PR using the [gh API](https://github.com/IntersectMBO/cardano-cli/pull/749),
+   > you will observe that the `PATCH` endpoint messes up existing metadata of the release (author, associated commit, etc.).
+   > So you HAVE to use the UI, as described in the previous item.
```
Comment (carbolymer) [inline RELEASING.md:157]:
```suggestion
   > So you HAVE to use the UI, as described above.
```

File: RELEASING.md [hunk 2]
```diff
@@ -146,6 +146,15 @@ Briefly speaking, it requires executing of the following steps:
     >
     >CHaP CI build can fail due to various reasons, like invalid haddock syntax.
     >Merging the release PR later allows easier adjusting of the tag to include the fix for the potential issues.
+1. If a GitHub release is automatically created by the CI, as visible on https://github.com/IntersectMBO/cardano-api/releases,
+   undraft the release by clicking the pen on the top right, then untick _Set as a pre-release_, and
+   finally select _Update release_.
+
+   >:warning: **GitHub bug**
+   >
+   > If you try to undraft a PR using the [gh API](https://github.com/IntersectMBO/cardano-cli/pull/749),
```
Comment (carbolymer) [inline RELEASING.md:155]: I'm not sure that I understand this link. What should I notice there?

**Review by carbolymer — APPROVED** [review]

File: RELEASING.md (hunk 2 above)
Comment (smelc) [reply RELEASING.md:155]: This was a copy/paste error, I meant https://docs.github.com/fr/rest/releases/releases?apiVersion=2022-11-28#update-a-release instead. I changed :+1:

File: RELEASING.md (hunk 1 above)
Comment (smelc) [reply RELEASING.md:157]: Changed :+1:

**Pull request merged**